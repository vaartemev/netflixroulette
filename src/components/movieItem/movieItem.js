import React from 'react';
import PropTypes from 'prop-types';
import { MovieTitle } from '../movieTitle/movieTitle';
import { Thumbnail } from '../thumbnail/thumbnail';

import './movieItem.scss';

export const MovieItem = ({ src, title, year, genre, handleOnClick }) => {
  return (
    <div className="movie" onClick={handleOnClick}>
      <Thumbnail src={src} className="movie__thumbnail" />
      <div className="movie__info">
        <MovieTitle className="movie__title" title={title} />
        <span className="movie__year">{new Date(year).getFullYear()}</span>
        <div className="movie__genre">{genre.join(', ')}</div>
      </div>
    </div>
  );
};

MovieItem.propTypes = {
  src: PropTypes.string,
  title: PropTypes.string,
  year: PropTypes.string,
  genre: PropTypes.array
};
