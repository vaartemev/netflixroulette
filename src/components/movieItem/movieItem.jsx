import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { MovieTitle } from '../movieTitle/movieTitle';
import { Thumbnail } from '../thumbnail/thumbnail';

import './movieItem.scss';

export const MovieItem = ({ id, src, title, year, genre, handleOnClick }) => {
  return (
    <Link to={`/film/${id}`}>
      <div className="movie" onClick={handleOnClick}>
        <Thumbnail src={src} className="movie__thumbnail" />
        <div className="movie__info">
          <MovieTitle className="movie__title" title={title} />
          <span className="movie__year">{new Date(year).getFullYear()}</span>
          <div className="movie__genre">{genre.join(', ')}</div>
        </div>
      </div>
    </Link>
  );
};

MovieItem.propTypes = {
  src: PropTypes.string,
  title: PropTypes.string,
  year: PropTypes.string,
  genre: PropTypes.array,
};
