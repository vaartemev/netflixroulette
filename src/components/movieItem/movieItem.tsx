import React from 'react';
import Link from 'next/link';
import { MovieTitle } from '../movieTitle';
import { Thumbnail } from '../thumbnail';

import './movieItem.scss';

interface MovieItemProps {
  id: number;
  src: string;
  title: string;
  year: string;
  genre: string;
  handleOnClick();
}

export const MovieItem: React.FunctionComponent<MovieItemProps> = ({
  id,
  src,
  title,
  year,
  genre,
  handleOnClick,
}): JSX.Element => {
  return (
    <Link href={`/movie?id=${id}`}>
      <div className="movie" onClick={handleOnClick} role="link" tabIndex="0">
        <Thumbnail src={src} className="movie__thumbnail" />
        <div className="movie__info">
          <MovieTitle className="movie__title" title={title} />
          <span className="movie__year">{new Date(year).getFullYear()}</span>
          <div className="movie__genre">{genre.join(', ')}</div>
        </div>
      </div>
    </Link>
  );
};
