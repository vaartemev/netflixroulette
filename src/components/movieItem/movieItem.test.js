import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MovieItem } from './movieItem';

describe('Movie Item', () => {
  const props = {
    src: 'string1',
    title: 'string2',
    year: 'string3',
    genre: [1, 2],
    handleOnClick: f => f()
  };

  const output = shallow(<MovieItem {...props} />);

  it('renders correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
