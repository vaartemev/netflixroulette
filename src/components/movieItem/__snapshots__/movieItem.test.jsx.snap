// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Item renders correctly 1`] = `
<Link
  to="/film/undefined"
>
  <div
    className="movie"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    <Thumbnail
      className="movie__thumbnail"
      src="string1"
    />
    <div
      className="movie__info"
    >
      <MovieTitle
        className="movie__title"
        title="string2"
      />
      <span
        className="movie__year"
      >
        NaN
      </span>
      <div
        className="movie__genre"
      >
        1, 2
      </div>
    </div>
  </div>
</Link>
`;
